File: C#/ByteBank/ByteBank.SistemaBanco.Core/SistemaInterno/ContaPoupanca.cs
namespace ByteBank.SistemaInterno
{
    public class ContaPoupanca : ContaCorrente
    {
        /// <summary>
        /// Construtor.
        /// </summary>
        /// <param name="titular">Titular da <see cref="ContaPoupanca"/>.</param>
        /// <param name="saldo">Saldo inicial da <see cref="ContaPoupanca"/>.</param>
        protected ContaPoupanca(string titular, double saldo) : base(titular, saldo)
        {
            TipoConta = "Conta Poupança";
            Tarifa = 0.05;
            Saldo = saldo - (saldo * Tarifa);
        }
    }
}

File: Java/ByteBank/src/com/codealligator/interno/ContaPoupanca.java
package com.codealligator.interno;

public class ContaPoupanca extends Conta {

    /*
    Construtor.
     */
    public ContaPoupanca(String titular, double saldo, double cpf, double numeroAgencia, double numeroConta) {
        super(titular, saldo, cpf, numeroAgencia, numeroConta);

        /*
        Seta a taxa/tarifa da Conta Poupança.
         */
        setTaxa(0.1);
    }
}
